Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,96 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            bool nonExistantCheckWarning = false;
+
+            try
+            {
+                if (externalBoundary.IIsSelfIntersecting())
+                {
+                    Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
```

Comment:
If we are returning some kind of objects anyway, would it be better to record a warning instead?

Reply:
Does `null` count as a object?

Reply:
If we exclude the invalid curves from the inner boundaries, and do not skip the entire calculation by returning a `PlanarSurface` built only with some if them, I think we do not always return `null`, don't we?

Reply:
Currently I'm returning null after every `RecordError` and using warnings if I thought it could be salvaged.

But could change the methods philosophy to avoid returning null and salvage more things?
- exclude invalid internal boundaries
- Cut the external boundery if it overlapps with a internal boundery

Reply:
I think I would lean towards @epignatelli - ruling out the incorrect curves with warning, using others to create the surface (with `NurbsCurve` not being checked at all, accepted by default). Not the most correct approach, but as a user I would expect something like that I think.